presenters/thread.rb

```diff
@@ -66,28 +68,41 @@ def to_hash(
         end
         case @thread.thread_type
         when "question"
-          endorsed_responses = responses.where(endorsed: true)
-          non_endorsed_responses = responses.where(endorsed: false)
-          endorsed_response_info = get_paged_merged_responses(
+          if merge_question_type_responses
+            response_info = get_paged_merged_responses(
             @thread._id,
-            endorsed_responses,
-            0,
-            nil,
-            recursive,
-            sorting_key_order
-          )
-          non_endorsed_response_info = get_paged_merged_responses(
-            @thread._id,
-            non_endorsed_responses,
+            responses,
             resp_skip,
             resp_limit,
             recursive,
             sorting_key_order
-          )
-          h["endorsed_responses"] = endorsed_response_info["responses"]
-          h["non_endorsed_responses"] = non_endorsed_response_info["responses"]
-          h["non_endorsed_resp_total"] = non_endorsed_response_info["response_count"]
-          h["resp_total"] = non_endorsed_response_info["response_count"] + endorsed_response_info["response_count"]
+            )
+            h["children"] = response_info["responses"]
+            h["resp_total"] = response_info["response_count"]
```

Comment:
do we not need any of these params in the resultant when `merge_question_type_responses` is provided for `question` type post?

```ruby
h["endorsed_responses"]
h["non_endorsed_responses"]
h["non_endorsed_resp_total"]
h["resp_total"] 
```

Reply:
Yes, this is the desired behavior to ensure that question type threads yield the same response structure as discussion type threads. To achieve this, all responses are intended to be contained within the `h["children"] = response_info["responses"] `  param. This ensures consistency across both thread types, aligning their response structures.